projects/packages/my-jetpack/_inc/components/product-card/recommendation-actions.tsx

```diff
@@ -0,0 +1,50 @@
+import { Button, getProductCheckoutUrl } from '@automattic/jetpack-components';
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import { useMemo } from 'react';
+import useProduct from '../../data/products/use-product';
+import { getMyJetpackWindowInitialState } from '../../data/utils/get-my-jetpack-window-state';
+import useMyJetpackConnection from '../../hooks/use-my-jetpack-connection';
+import styles from './style.module.scss';
+
+const useUpsellLinks = ( slug: string ) => {
+	const { detail } = useProduct( slug );
+	const { isUserConnected } = useMyJetpackConnection();
+
+	return useMemo( () => {
+		const { adminUrl, siteSuffix } = getMyJetpackWindowInitialState();
+		const purchaseUrl = getProductCheckoutUrl(
+			detail.wpcomProductSlug,
+			siteSuffix,
+			`${ adminUrl }?page=my-jetpack`,
+			isUserConnected
+		);
+		const interstitialUrl = `#/add-${ slug }`;
+
+		return { purchaseUrl, interstitialUrl };
+	}, [ slug, detail.wpcomProductSlug, isUserConnected ] );
+};
+
+const RecommendationActions = ( { slug }: { slug: string } ) => {
+	const { purchaseUrl, interstitialUrl } = useUpsellLinks( slug );
+
+	return (
+		<div className={ styles.actions }>
+			<div className={ clsx( styles.buttons, styles.upsell ) }>
+				<Button size="small" href={ purchaseUrl }>
+					{ __( 'Purchase', 'jetpack-my-jetpack' ) }
+				</Button>
+				<Button
+					className={ styles.recommendationLink }
+					size="small"
+					variant="link"
+					href={ interstitialUrl }
+				>
+					{ __( 'Start for free', 'jetpack-my-jetpack' ) }
```

Comment:
Hmm, not every product has the option to "Start for Free" do they?  For example, Backup?

See Screenshot:
![Screen Shot 2024-07-16 at 10 54 40](https://github.com/user-attachments/assets/d6d3f2cc-67ed-4293-841a-b120e6d748d5)

I don't think it will be good for user experience when we show a "Start for free" link, but then when you click it, there is no free option, but rather it's asking you to pay instead, like what happens with Backup  Unless maybe this is something that is soon upcoming? Adding a free option for All products? Did I see that somewhere? 🤔

Reply:
Yeah, that's correct, I'm developing the mechanism to recognize products that offer free start options.